server/log_rpc_server.go

```diff
@@ -78,12 +78,6 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
 	}
 	leaves := depointerify(req.Leaves)
 
-	// TODO(al): Hasher must be selected based on log config.
-	th, _ := merkle.Factory(merkle.RFC6962SHA256Type)
```

Comment:
I don't have a strong opinion, but I don't think I completely understand what the implications might be either.

This is straying a bit out of the purpose of this PR, would you mind reverting and maybe putting a separate PR together for this change? I think this should be discussed with more people in the team.